### ISSUE TITLE
Review and Correct WaitGroup Usage in `Run` Function of `volumeprovider/app.go`

### ISSUE THREAD

**Issue by so-sahu** [ironcore-dev/ceph-provider#572]
### Description:

In the `Run` function of the `volumeprovider`'s `app.go` file, there is a `sync.WaitGroup` (`wg`) being used to manage the execution of multiple goroutines. However, the current implementation does not appear to make logical sense:

The `wg.Add(1)` statement is being called before starting each goroutine.
`wg.Done()` is deferred within each goroutine to signal completion.

However, the `wg.Wait()` method, which should be used to block the main function until all goroutines have completed, is not being called anywhere within the `Run` function.

This usage of `WaitGroup` without a corresponding `wg.Wait()` does not serve any practical purpose and could potentially lead to unpredictable behavior, especially in concurrent execution scenarios. 

### Suggested Solution:

**Option 1:** If the intention is to wait for all goroutines to complete before exiting the `Run` function, then a `wg.Wait()` should be added at the appropriate place in the code.
  
**Option 2:** If waiting for goroutines to finish is not necessary, then the `WaitGroup` can be removed entirely to simplify the code.

**Option 3.** For reference,  you can view the similar implementation in the `libvirt-provider` project [here](https://github.com/ironcore-dev/libvirt-provider/blob/c23ba3e43fab156ad688e2eeb0661d953a9959a9/cmd/libvirt-provider/app/app.go#L385).

### Expected Behavior:

The code should either properly implement the `WaitGroup` mechanism to wait for all goroutines to finish or remove the `WaitGroup` if it is not needed.

### Additional Context:

This issue might affect the reliability and predictability of the `Run` function, especially in a production environment with concurrent operations. Proper synchronization is crucial to ensure the correct execution flow.

### Action Items:

- [ ] Review the `Run` function in `volumeprovider`'s `app.go`.
- [ ] Determine the necessity of waiting for all goroutines.
- [ ] Either implement `wg.Wait()` appropriately or remove the `WaitGroup` usage.
- [ ] Test the changes to ensure that the function behaves as expected.

**Comment by lukasfrank** [comment]
Thanks a lot for the issue, it's a total valid point that `wg.Wait()` is missing. It would be great, if you can take care of it.
